src/main/kotlin/snyk/container/ui/BaseImageRemediationDetailPanel.kt

```diff
@@ -0,0 +1,292 @@
+package snyk.container.ui
+
+import com.intellij.ide.util.PsiNavigationSupport
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiDocumentManager
+import com.intellij.psi.PsiFile
+import com.intellij.ui.components.labels.LinkLabel
+import com.intellij.uiDesigner.core.GridConstraints
+import com.intellij.uiDesigner.core.GridLayoutManager
+import com.intellij.uiDesigner.core.Spacer
+import icons.SnykIcons
+import io.snyk.plugin.Severity
+import io.snyk.plugin.getKubernetesImageCache
+import snyk.container.BaseImageInfo
+import snyk.container.BaseImageVulnerabilities
+import snyk.container.ContainerIssuesForImage
+import snyk.container.KubernetesWorkloadImage
+import java.awt.Color
+import java.awt.Component
+import java.awt.Dimension
+import java.awt.Font
+import java.awt.Insets
+import javax.swing.JComponent
+import javax.swing.JLabel
+import javax.swing.JPanel
+import javax.swing.JTextArea
+
+class BaseImageRemediationDetailPanel(
+    private val project: Project,
+    private val imageIssues: ContainerIssuesForImage
+) : JPanel() {
+
+    private val targetImages: List<KubernetesWorkloadImage> = getKubernetesImageCache(project)
+        ?.getKubernetesWorkloadImages()
+        ?.filter { it.image == imageIssues.imageName }
+        ?: emptyList()
+
+    init {
+        this.layout = GridLayoutManager(10, 1, Insets(20, 10, 20, 20), -1, 0)
```

Comment:
Maybe we should start putting these Grid-Magic-Numbers into a central class as constants instead of redefining it in all panels and products. The `20` is used all over the UI code for example and could be something like `STD_VERTICAL_SPACING` or so.

Reply:
We definitely need to review and refactor whole UI part of the plugin. the current way to display is came form legacy and we need to change it.
But that a way out of this ticket scope.
Would suggest to do kind of hackathon for UI and Tree model refactoring within nearest Small Hands.